### PR TITLE
Merge fixes for PS-9144 and PS-9214 into 8.4 tree.

### PR DESCRIPTION
**Pull request by dlenev**
See individual commits for detailed description.

### CODE REVIEW

**Review by satya-bodapati — APPROVED** [review]
Changes same as 8.0. LGTM!

File: mysql-test/suite/innodb/t/percona_alter_debug.test [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Tests in this file use both debug-only facilities as well as debug sync point.
```
Comment (percona-ysorokin) [inline mysql-test/suite/innodb/t/percona_alter_debug.test]: @dlenev in 8.4, please move percona-specific InnoDB tests to "percona_innodb" test suite.
Also, rename the test to just `alter_debug`.

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

File: mysql-test/suite/innodb/t/percona_alter_debug.test (hunk 1 above)
Comment (dlenev) [reply mysql-test/suite/innodb/t/percona_alter_debug.test]: Done.

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Comment by dlenev** [comment]
Jenkins results look OK to me: https://ps80.cd.percona.com/view/8.0%20parallel%20MTR/job/percona-server-8.x-pipeline-parallel-mtr/262/

**Pull request merged**